test/test_helpers/controller_helper.rb

```diff
@@ -9,28 +9,37 @@ module ControllerHelper
   #   setup { @touhou = create(:tag, name: "touhou") }
   #   should respond_to_search(name: "touhou").with { @touhou }
   #
-  def respond_to_search(search_params)
-    RespondToSearchMatcher.new(search_params)
+  def respond_to_search(search_params, other_params: {})
+    RespondToSearchMatcher.new(search_params, other_params)
   end
 
-  class RespondToSearchMatcher < Struct.new(:params)
+  class RespondToSearchMatcher < Struct.new(:params, :other_params)
     def description
       "should respond to a search for #{params}"
     end
 
     def matches?(subject, &block)
-      search_params = { search: params }
+      search_params = other_params.merge({ search: params })
       expected_items = @test_case.instance_eval(&@expected)
 
       @test_case.instance_eval do
         # calls e.g. "wiki_pages_path" if we're in WikiPagesControllerTest.
         index_url = send("#{subject.controller_path}_path")
-        get index_url, as: :json, params: search_params
+        # Allows for different authorization levels to be used, instead of just anonymous
+        if CurrentUser.user.present?
```

Comment:
The user should be passed explicitly instead of implicitly relying on `CurrentUser`.

Reply:
Yeah, I hadn't really thought of that. Though that would also necessitate passing in the the user for every single shoulda call that are done by levels other than anonymous, which are all or most in some cases.